programs/client/Client.cpp

```diff
@@ -361,6 +358,8 @@ try
         showClientVersion();
     }
 
+    createClientContext();
```

Comment:
Here the client context is created after processing the configuration and query parameters but before connecting to the server or parsing any queries from user.

Reply:
I added a comment to the [code](https://github.com/ClickHouse/ClickHouse/blob/0cf0437196dfe4ee0f489ecc040b71e42e1f1a22/programs/local/LocalServer.cpp#L513-L516) to make the idea more clear.